File: UVA10925/UVA10925.py

bill = 1
while True:
    N_str, F_str = input().split(" ")
    N = int(N_str)
    F = int(F_str)
    if N == F and N == 0:
        break

    sum_ = 0
    for _ in range(N):
        sum_ += int(input())
    
    print("Bill #{} costs {}: each friend should pay {}".format(bill, sum_, sum_ //F))
    print()
    bill += 1

File: UVA10925/UVA10925.java
import java.io.*;
import java.util.*;
import java.math.BigInteger;

public class UVA10925 {
        public static void main(String[] argv){
            int N, F, i, turn = 1;
            Scanner scan = new Scanner(System.in);
            while (true) {
        
                N = scan.nextInt();
                F = scan.nextInt();
                if (N == 0 && F == 0){
                    break;
                }   
                BigInteger price, sum = BigInteger.ZERO;
                for (i = 0; i < N; i++){
                    price = scan.nextBigInteger();
                    sum = sum.add(price);
                }
                System.out.printf("Bill #%d costs %d: each friend should pay %d\n\n", turn, sum, sum.divide(BigInteger.valueOf(F)));
                turn ++;
            }
        }
}
